include/deal.II/dofs/dof_tools.h

```diff
@@ -1760,6 +1760,25 @@ namespace DoFTools
                       const bool            invert_vertex_mapping  = false);
 
   /**
+   *  Same as above but allows boundary dofs on blocks to be excluded individually
+   *
+   * @arg <tt>exclude_boundary_dofs</tt>:  for each patch of cells around a
+   * vertex, collect all of the interior degrees of freedom of the patch and
+   * disregard those on the boundary of the patch if the boolean value for
+   * the corresponding block in the BlockMask is false.
+   */
+  template <typename DoFHandlerType>
+  std::vector<unsigned int>
+  make_vertex_patches(SparsityPattern      &block_list,
+                      const DoFHandlerType &dof_handler,
+                      const unsigned int    level,
+                      const BlockMask      &exclude_boundary_dofs  = BlockMask(),
```

Comment:
We did it because we thought it made more sense for our purposes.  I have added an example in the header file of why this new functionality could be useful that should help clarify things a bit.  

Also, we had originally added the extra argument to the existing function, rather than adding a new one as we now do, but we decided it made more sense to add the new one since the original functionality was a subset of what is now possible with our functionality.

Let me know if you want any more explanations or changes!